### ISSUE TITLE
Add Early Stopping and test dataset for better training

### ISSUE THREAD

**Issue by MohamedAliRashad** [stochasticai/xturing#85]
A better training experience will be if the dataset passed to the `finetune` method had a test dataset in it and there was an `EarlyStopping` functionality in the training process that automatically stops the training process if the loss is not going down anymore.

**Comment by RomaA2000** [comment]
Hi @MohamedAliRashad,

We are working on adding this feature, thank you for pointing out!
Now you can use .num_train_epochs of config to modify finetuning process length.

**Comment by StochasticRomanAgeev** [comment]
Hi @MohamedAliRashad,

We are working on adding this feature, thank you for pointing out!
Now you can use .num_train_epochs of config to modify finetuning process length.
